server/CGameHandler.cpp

```diff
@@ -1897,6 +1897,14 @@ void CGameHandler::showBlockingDialog( BlockingDialog *iw )
 	sendToAllClients(iw);
 }
 
+void CGameHandler::showMonolithDialog( MonolithDialog *iw )
+{
+	auto dialogQuery = make_shared<CMonolithDialogQuery>(*iw);
+	queries.addQuery(dialogQuery);
+	iw->queryID = dialogQuery->queryID;
+	sendToAllClients(iw);
```

Comment:
As you directly accessing player interface on client side 99.9% crash in multiplayer.